tests/watch.test.tsx

```diff
@@ -96,4 +102,50 @@ describe('watch', () => {
 
     reference.value = 'Update'
   })
+  it('should support promise watchers', async () => {
+    const reference = proxy({ value: 'Example' })
+
+    const callback = vi.fn()
+
+    const waitPromise = wait(1000)
```

Comment:
1000ms is too long. Ideally, we want to have fake timers. We do something in `jotai`, but not in `valtio`. Hmm.

Maybe, it's not very difficult in this case?

```ts
let resolve
const waitPromise = new Promise((r) => { resolve = r })
...
resolve()
```

Reply:
Updated with fake timers from vitest - does that suffice? Also just reduced these to `0` but kept the timeout to make sure they are scheduled as a macrotask

Reply:
> Updated with fake timers from vitest - does that suffice?

Yes! In such a case, we don't need to make it `0`, right? `1000` would be more meaningful. Or, maybe bigger. I set something like 30 sec somewhere, which makes it easy to know if the fake timer works as expected.

Reply:
You're right